src/Microsoft.Framework.PackageManager/SourceControl/SourceControlProviderFactory.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.PackageManager.SourceControl
+{
+    internal static class SourceControlProviderFactory
+    {
+        // TODO: When we have a second source control provider extract the common API in a base class/interface
```

Comment:
Are there bugs filed for specific providers we're interested in? `hg` ? `tfvc` ?

Also, should we consider allowing users to install custom providers?

[All these are just things to consider for the future, but we should capture them in bugs]